### PR TITLE
Task/parent component works

### PR DESCRIPTION
**Pull request by EduardoVaca**
#### What's this PR do?
Whole CRUD of a work

#### Where should the reviewer start?
In works folder

#### How should this be manually tested?
All CRUD of works

#### Any background context you want to provide?
Fill DB first of
Iguala
ArtTypes
Status
Clients
Contacts And WorkTypes

This template was adapted ~stolen~ from [Quickleft/Sprint.ly](https://quickleft.com/blog/pull-request-templates-make-code-review-easier/)

### CODE REVIEW

File: src/app/general-user/works/work-form.component.ts [hunk 1]
```diff
@@ -19,23 +31,288 @@ import { CustomToastService } from '../../shared/toast/custom-toast.service';
...
 
   public constructor(private httpService: HttpService, private toaster: CustomToastService) { }
+
   /**
   * Builds the component for first time each time when it's called.
   *   - Initialize the form depending if the new or update work form is called.
   *   - Use an auxiliary variable to select a default value for the dropdown used in the form.
   **/
   public ngOnChanges() {
+    console.log('CALLING');
```
Comment (JuanGro) [inline src/app/general-user/works/work-form.component.ts]: Could you hide this?

File: src/app/general-user/works/works.component.ts [hunk 2]
```diff
@@ -35,15 +64,203 @@ export class WorksComponent implements OnInit {
...
+  * Loads all the works from the get method in httpService to use it the work attribute of the current component.
+  * Params:
+  *   - url: The url where the service will comunicate to get the Work objects.
+  **/
+  private loadWorksList(url: string) {
+    this.httpService.getObject(url)
+                    .map((data: any) => data.json())
+                    .subscribe(worksListJSON => {
+                      this.worksList = [];
+                      console.log(worksListJSON);
```
Comment (JuanGro) [inline src/app/general-user/works/works.component.ts]: Could you hide this?

File: src/app/general-user/works/works.component.ts [hunk 3]
```diff
@@ -35,15 +64,203 @@ export class WorksComponent implements OnInit {
...
+                      for (let workJSON of worksListJSON) {
+                        this.worksList.push(new Work(workJSON));
+                      }
+                      console.log(this.worksList);
```
Comment (JuanGro) [inline src/app/general-user/works/works.component.ts]: and this

File: src/app/general-user/works/works.component.ts [hunk 4]
```diff
@@ -35,15 +64,203 @@ export class WorksComponent implements OnInit {
...
+                    .map((data: any) => data.json())
+                    .subscribe(igualasListJSON => {
+                      // Creates Iguala objects from JSON.
+                      this.igualasList = [];
+                      for (let igualaJSON of igualasListJSON) {
+                        this.igualasList.push(new Iguala(igualaJSON));
+                      }
+                    },
+                      err => {
+
```
Comment (JuanGro) [inline src/app/general-user/works/works.component.ts]: Could you show a toaster with the error?

File: src/app/general-user/works/works.component.ts [hunk 5]
```diff
@@ -35,15 +64,203 @@ export class WorksComponent implements OnInit {
...
+                    .map((data: any) => data.json())
+                    .subscribe(contactsListJSON => {
+                      // Creates Contact objects from JSON.
+                      this.contactsList = [];
+                      for (let contactJSON of contactsListJSON) {
+                        this.contactsList.push(new Contact(contactJSON));
+                      }
+                    },
+                      err => {
+                        // Call of toast
```
Comment (JuanGro) [inline src/app/general-user/works/works.component.ts]: Could you show a toaster with the error?

File: src/app/general-user/works/works.spec.ts [hunk 6]
```diff
@@ -0,0 +1,222 @@
...
+        });
+    });
+
+    describe('Initialization of variable for parent works component', () => {
+        /**
+        * Tests that the page title is correct.
+        **/
+        it('should show the work component with correct attributes', () => {
+            fixtureParent.detectChanges();
+            expect(component.title).toContain('Lista de Trabajos');
```
Comment (JuanGro) [inline src/app/general-user/works/works.spec.ts]: Please change ".toContain()" to ".toBe()" or "toEqual()" for all this tests.

File: src/app/general-user/works/work-form.component.ts (hunk 1 above)
Comment (EduardoVaca) [inline src/app/general-user/works/work-form.component.ts]: Done

File: src/app/general-user/works/works.component.ts (hunk 2 above)
Comment (EduardoVaca) [inline src/app/general-user/works/works.component.ts]: done

File: src/app/general-user/works/works.spec.ts (hunk 6 above)
Comment (EduardoVaca) [inline src/app/general-user/works/works.spec.ts]: done

File: src/app/general-user/works/works.component.ts (hunk 5 above)
Comment (EduardoVaca) [inline src/app/general-user/works/works.component.ts]: done

File: src/app/general-user/works/works.component.ts (hunk 4 above)
Comment (EduardoVaca) [inline src/app/general-user/works/works.component.ts]: done

**Pull request merged**